demo/run_tests.sh

```diff
@@ -1,18 +1,36 @@
-$1 -s --headless --path $PWD addons/gut/gut_cmdln.gd | ( no_error="false"; while read -r line
-do
-    echo "$line"
-    if echo "$line" | grep -q "Passing tests     0"; then
-        echo "ERROR: No tests passed !"
-        exit 1
+$1 -s --headless --path $PWD addons/gut/gut_cmdln.gd | (
+    tests=0
+    passing=0
+
+    while read -r line
+    do
+        echo "$line"
+
+        # Capture the number of tests
+        if echo "$line" | grep -q "Tests"; then
+            tests=$(echo "$line" | awk '{print $2}')
+        fi
+
+        # Capture the number of passing tests
+        if echo "$line" | grep -q "Passing"; then
+            passing=$(echo "$line" | awk '{print $2}')
+        fi
+    done
+
+    if [[ "$tests" -eq 0 ]]; then
+       echo "ERROR: No tests were found."
+       exit 1
     fi
-    if echo "$line" | grep -q "Failing tests     0"; then
```

Comment:
This does not work any longer in latest GUT. A better way is to extract the number of passing tests vs total tests and ensure they are the same.